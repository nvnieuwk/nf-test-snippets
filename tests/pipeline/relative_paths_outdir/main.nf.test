nextflow_pipeline {

    name "Get all relative paths of the output directory without checking md5sums"
    script "main.nf"

    tag "relative_paths"

    test("Get all relative paths of the output directory without checking md5sums") {

        when {
            params {
                outdir = outputDir
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten()
                ).match("relative paths") }
            )
        }

    }

}

def getRecursiveFileNames(fileOrDir, outputDir) {
    if(file(fileOrDir.toString()).isDirectory()) {
        return fileOrDir.list().collect { getRecursiveFileNames(it, outputDir) }
    }
    return fileOrDir.toString().replace("${outputDir}/", "")
}

