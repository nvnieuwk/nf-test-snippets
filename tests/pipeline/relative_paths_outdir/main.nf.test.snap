{
    "relative paths": {
        "content": [
            [
                "split/loremaa.txt",
                "split/loremab.txt",
                "split/loremac.txt",
                "split/loremad.txt",
                "split/loremae.txt",
                "split/loremaf.txt",
                "split/loremag.txt",
                "split/loremah.txt",
                "split/loremai.txt",
                "split/loremaj.txt",
                "split/loremak.txt",
                "split/sub_folder/loremaa.txt",
                "split/sub_folder/loremab.txt",
                "split/sub_folder/loremac.txt",
                "split/sub_folder/loremad.txt",
                "split/sub_folder/loremae.txt",
                "split/sub_folder/loremaf.txt",
                "split/sub_folder/loremag.txt",
                "split/sub_folder/loremah.txt",
                "split/sub_folder/loremai.txt",
                "split/sub_folder/loremaj.txt",
                "split/sub_folder/loremak.txt"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-28T10:07:37.027816463"
    }
}